### PR TITLE
Improved logs in docker auth extensions

### PR DESCRIPTION
**Pull request by tharindulak**

### CODE REVIEW

File: components/docker-auth/cmd/authn/authentication.go [hunk 1]
```diff
@@ -43,237 +43,268 @@ const (
...
+	log.Printf("[%s] Subject : %s\n", execId, sub)
 
 	if sub != username {
-		log.Println("Username does not match with subject in JWT")
+		log.Printf("[%s] Username(%s) does not match with subject(%s) in JWT\n", execId, username, sub)
 		os.Exit(extension.ErrorExitCode)
 	}
 
-	certificateInUse, err := readCert(idpCertEnvVar)
+	certificateInUse, err := readCert(idpCertEnvVar, execId)
```
Comment (nadundesilva) [inline components/docker-auth/cmd/authn/authentication.go]: Don't we need to handle this error ?

File: components/docker-auth/cmd/authn/authentication.go [hunk 2]
```diff
@@ -43,237 +43,268 @@ const (
...
 
 func main() {
+	err := os.MkdirAll("/extension-logs", os.ModePerm)
+	if err != nil {
+		log.Println("Error creating the file :", err)
+	}
 	file, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
 	if err != nil {
-		log.Fatalf("Error opening file: %v", err)
+		log.Printf("Error opening file: %s\n", err)
```
Comment (nadundesilva) [inline components/docker-auth/cmd/authn/authentication.go]: You can move this log inside the if condition in line 111 and remove this if block

Comment (nadundesilva) [inline components/docker-auth/cmd/authn/authentication.go]: You can move this log into the if block in line 111 and remove this if block.

File: components/docker-auth/cmd/authn/authentication.go [hunk 3]
```diff
@@ -43,237 +43,268 @@ const (
...
 	}
 
-	certificateInUse, err := readCert(idpCertEnvVar)
+	certificateInUse, err := readCert(idpCertEnvVar, execId)
 
 	if iss == authTokenIssuerEnvVar {
-		certificateInUse, err = readCert(dockerAuthCertEnvVar)
+		certificateInUse, err = readCert(dockerAuthCertEnvVar, execId)
 	}
 	if err != nil {
```
Comment (nadundesilva) [inline components/docker-auth/cmd/authn/authentication.go]: Let's move this inside the above if block

File: components/docker-auth/cmd/authn/authentication.go (hunk 1 above)
Comment (tharindulak) [reply components/docker-auth/cmd/authn/authentication.go]: Fixed

File: components/docker-auth/pkg/extension/execIDGenerator.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package extension
```
Comment (nadundesilva) [inline components/docker-auth/pkg/extension/execIDGenerator.go]: Please add license header

File: components/docker-auth/test/data.sql [hunk 5]
```diff
@@ -1,8 +1,9 @@
 USE CELLERY_HUB;
```
Comment (nadundesilva) [inline components/docker-auth/test/data.sql]: Let's add license header here as well.

File: components/docker-auth/pkg/extension/execIDGenerator.go (hunk 4 above)
Comment (tharindulak) [reply components/docker-auth/pkg/extension/execIDGenerator.go]: Fixed in #85

File: components/docker-auth/test/data.sql (hunk 5 above)
Comment (tharindulak) [reply components/docker-auth/test/data.sql:1]: Fixed by #178

**Pull request merged**